terraso_backend/apps/soil_id/graphql/soil_id/resolvers.py

```diff
@@ -0,0 +1,272 @@
+# Copyright © 2024 Technology Matters
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see https://www.gnu.org/licenses/.
+
+from typing import Optional
+
+from soil_id.us_soil import list_soils, rank_soils
+
+from apps.soil_id.graphql.soil_data import DepthInterval
+from apps.soil_id.graphql.soil_id.sample_data import (
+    dummy_data_matches,
+    dummy_location_matches,
+)
+from apps.soil_id.graphql.soil_id.schema import (
+    DataBasedSoilMatch,
+    DataBasedSoilMatches,
+    EcologicalSite,
+    LABColorInput,
+    LandCapabilityClass,
+    LocationBasedSoilMatch,
+    LocationBasedSoilMatches,
+    SoilIdDepthDependentData,
+    SoilIdInputData,
+    SoilIdSoilData,
+    SoilInfo,
+    SoilMatchInfo,
+    SoilSeries,
+)
+from apps.soil_id.models.depth_dependent_soil_data import DepthDependentSoilData
+from apps.soil_id.models.soil_data import SoilData
+
+
+def resolve_texture(texture: str | float):
+    if not isinstance(texture, str) or texture == "":
```

Comment:
```suggestion
    if not isinstance(texture, str) or not texture:
```

Reply:
same as above, i like being explicit about what i am testing for